### PR TITLE
Adds a pizza producing chem

### PR DESCRIPTION
**Pull request by GuyonBroadway**
:cl: GuyonBroadway and Neetdev
add: Mix a da oil, da salt, da tomato-juice and a da flour and heat to 450 and BAM, a delicious a pizza just like a mamma used to make!
/:cl:

https://youtu.be/DJZzGJ9SYzA

https://youtu.be/gUqH6Weyr2M

Pizza grenades, literally the only reason why.

### CODE REVIEW

**Comment by Arzion5** [comment]
Italian millitary-grade grenades. I support it because it's funny

**Comment by PopNotes** [comment]
You only get my approval if you call it pizzafication

**Comment by Pyko1** [comment]
Seems bad, could prolly make tons of pizzas and grind them for cheap nutriment, tomatojuice, vitamin, oculine, omnizine, and doctorsdelight.

Maybe make them spawn some crap pizzas that are mostly just sugar or something.

**Review by ghost — DISMISSED** [review]

**Comment by KayeArray** [comment]
now i want pizza cunt +1

**Comment by Pyko1** [comment]
I still think the pizza it makes should be a vender tier pizza,

Proper pizzas are pretty much the best food a chef can make, and at least on a quick glance the chemists could make a far more pizzas than the chef from starting reagents alone.

I mean whats the point of kitchen/chef if chemistry can make more food with less mats and effort?
Just make it spawn crap tier pizzas that are mostly sugar/tomatosauce/salt

**Comment by KayeArray** [comment]
@Pyko1 how about 9 times out of 10 it makes ""delicious"" corn potato pizza instead.

**Comment by Pyko1** [comment]
yeah that sounds good, i was just talking about adding it.

Its neurotoxing to anyone else than carbonhell, it releases fumes that can melt rwalls, 95% of people who see it get a corn-potato pizza overlay filling their screen that prevents them from seeing anything else (possibly during multiple rounds)

**Comment by KayeArray** [comment]
@Pyko1 perhaps a more conservative forced to vomit until you dry heave + max liver damage instead?

**Comment by Pyko1** [comment]
Even tho I kinda want to add corn-potato pizzas, they are like ark they spam messages to everyone.

You are overwhelmed by an absolutely disgusting stench coming from north-west.

**Comment by Pyko1** [comment]
This is corn-potato pizza, first time in your life you regret not having your eyes, or head surgically removed.

**Comment by Pyko1** [comment]
also has chance to make request atmos to flood the station with plasma

**Comment by tortellinitony** [comment]
yea don't let it spawn any pizzas that have op shit in them like omnizine and make sure it doesn't give an insane amount of nutriment per unit of pizza chem..otherwise a pretty good idea

**Comment by tortellinitony** [comment]
also mob react so you can fill the clown with this and light him on fire

**Comment by dreamcrab** [comment]
make it a cheap frozen supermarket-style garbo pizza and im on board with this

**Comment by ghost** [comment]
yes

**Comment by Spacedong** [comment]
epic

**Comment by ghost** [comment]
it sounds dumb and cool but make it shit-tier pizzas with negative effects
from spaghettification to pizzafication

**Comment by GuyonBroadway** [comment]
Thats kinda outside the pr scope, what it does do is randomly generate any pizza so if somone did code some shitty pizzas it would spawn those too.

**Comment by ghost** [comment]
yea ok
+1 then

**Comment by GuyonBroadway** [comment]
Neat thats three +1s.

File: hippiestation/code/modules/food_and_drinks/recipes/food.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/datum/chemical_reaction/pizzaification
+	name = "Pizzaification"
+	id = "pizzaification"
+	required_reagents = list("tomatojuice" = 10, "oil"  = 5, "flour"=20, "sodium"=5)
+	required_temp = 450
+	mob_react = FALSE
+
+/datum/chemical_reaction/pizzaification/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
```
Comment (JohnGinnane) [inline hippiestation/code/modules/food_and_drinks/recipes/food.dm:10]: Trying to find where `created_volume` is set but can't.. I'd just be concerned that the base ingredients, 10 tomato juice + 5 oil + 20 flour + 5 sodium end up creating, 40 different pizzas

Comment (GuyonBroadway) [reply hippiestation/code/modules/food_and_drinks/recipes/food.dm:10]: Fear not for I test my code and it only makes 1 pizza per 40u

Comment (JohnGinnane) [reply hippiestation/code/modules/food_and_drinks/recipes/food.dm:10]: Oh, that's handy

**Review by JohnGinnane — APPROVED** [review]

**Comment by JohnGinnane** [comment]
Three +1s to merge?

**Comment by Pyko1** [comment]
not sure really as some heads have agreed that it spawning from all pizzas might have issues, but no one has really addressed about it.

**Comment by GuyonBroadway** [comment]
Yes it takes 3

**Comment by Pyko1** [comment]
But i guess we can just nerf it later

**Pull request merged**